Assets/_Functionalities/AssetBundles/Scripts/AssetBundleLoader.cs

```diff
@@ -0,0 +1,148 @@
+
+        private void Start()
+        {
+            //ProjectData.Current.OnDataChanged.AddListener(OnProjectDataChanged);
+        }
+
+        private void OnProjectDataChanged(ProjectData projectData)
+        {
+            //we want only the original asset here and not an instantiation
+            LoadAssetFromAssetBundle(bundleName, prefabName, Vector3.zero, asset =>
+            {
+                HierarchicalObjectLayerGameObject layerObject = asset.GetComponent<HierarchicalObjectLayerGameObject>();
+                if (layerObject != null)
+                {                
+                    projectData.PrefabLibrary.AddObjectToPrefabGroupRuntime("ObjectenBibliotheek", layerObject);
+                    if(!hierarchicalObjectLayerGameObjects.Contains(layerObject))
+                        hierarchicalObjectLayerGameObjects.Add(layerObject);
+                }
+            });
+        }
+
+        public void LoadAssetFromAssetBundle(string bundleName, string fileName, Vector3 position, Action<GameObject> onLoaded)
+        {
+            string path = Path.Combine(Application.streamingAssetsPath, bundleName);
+            StartCoroutine(GetAssetBundle(path, bundle =>
+            {
+                string[] names = bundle.GetAllAssetNames();
+                foreach (string n in names)
+                {
+                    if (Path.GetFileName(n) == fileName.ToLower())
+                    {
+                        GameObject asset = bundle.LoadAsset<GameObject>(n);
+                        if (asset != null)
+                        {
+#if UNITY_EDITOR
+                            FixShadersForEditor(asset);
+#endif
+                            onLoaded(asset);
+                        }
+                        bundle.Unload(false);
+                        return;
+                    }
+                }
+            }));
+        }
+
+        public void CreateAssetFromAssetBundle(string bundleName, string fileName, Vector3 position, Action<GameObject> onLoaded)
+        {
+            string path = Path.Combine(Application.streamingAssetsPath, bundleName);
+            StartCoroutine(GetAssetBundle(path, bundle =>
+            {
+                string[] names = bundle.GetAllAssetNames();
+                foreach (string n in names)
+                {
+                    if (Path.GetFileName(n) == fileName.ToLower())
+                    {
+                        GameObject asset = bundle.LoadAsset<GameObject>(n);
+                        if (asset != null)
+                        {
+                            GameObject model = Instantiate(asset);
+#if UNITY_EDITOR
+                            FixShadersForEditor(model);
+#endif
+                            onLoaded(model);
+                        }
+                        bundle.Unload(false);
+                        return;
+                    }
+                }
+            }));
+        }
+
+        private void FixShadersForEditor(GameObject asset)
```

Comment:
als dit alleen in de editor is, kan je er dan #if Unity_Editor omheen zetten?

Reply:
staat er al omheen?